src/backends/github/API.js

```diff
@@ -174,9 +175,23 @@ export default class API {
     return unpublishedPromise;
   }
 
+  buildUnpublishedPRSearchQuery(branch) {
+    return `repo:${ this.repo } type:pr state:open head:${ branch }`;
+  }
+
   listUnpublishedBranches() {
     console.log("%c Checking for Unpublished entries", "line-height: 30px;text-align: center;font-weight: bold"); // eslint-disable-line
+
     return this.request(`${ this.repoURL }/git/refs/heads/cms`)
+    .then(branches => filterPromises(branches, (branch) => {
+      const branchName = branch.ref.substring("/refs/heads/".length - 1);
+
+      // Get PRs with a `head` of `branchName`. Note that this is a
+      // substring match, so we need to check that the `head.ref` of
+      // at least one of the returned objects matches `branchName`.
+      return this.request(`${ this.repoURL }/pulls?head=${ branchName }&state=open`)
+        .then(prs => prs.reduce((exists, pr) => exists || pr.head.ref === branchName, false));
```

Comment:
Consider using `prs.some` instead of `prs.reduce`, more self-documenting that way, and a tad shorter.